### DISCUSSION TITLE
Mixing C# 11 and .NET 6

### DISCUSSION THREAD

**Discussion by DavidArno** [dotnet/csharplang#6737] [Q&A] [answered]
As .NET 7 is a short-term support release and will go out of support a year before .NET 6 does. Therefore, there's a compelling business case for not using .NET 7 for "enterprise" products, ie ones where the customer prefers infrequent product releases.

Is there any official guidance from the C# language team, or MS in general, as to whether any C# 11 features are supported for use with .NET 6, or is this all "not supported; use at your own risk" territory?

Thanks.

**Comment by HaloFour** [comment, marked as answer]
Using a downlevel version of the runtime is not supported, regardless of the support terms.  For those that want LTS, they'll need to wait until the next release of C# that corresponds with an LTS release of the runtime.

This is also how Java manages releases.

**Reply by DavidArno** [reply]
Do you know if this is explicitly documented anywhere, or just inferred from comments regarding previous releases?

**Reply by HaloFour** [reply]
Documented here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version

> [C# 11](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11) is supported only on .NET 7 and newer versions.

**Reply by DavidArno** [reply]
Thanks, @HaloFour. Much appreciated.
